File: binding/src/libqtyperevision.cpp
#include <QDataStream>
#include <QTypeRevision>
#include <qtyperevision.h>
#include "libqtyperevision.h"
#include "libqtyperevision.hxx"

QTypeRevision* QTypeRevision_new(const QTypeRevision* other) {
	 return new QTypeRevision(*other);
}

QTypeRevision* QTypeRevision_new2(QTypeRevision* other) {
	 return new QTypeRevision(*other);
}

QTypeRevision* QTypeRevision_new3() {
	 return new QTypeRevision();
}

QTypeRevision* QTypeRevision_new4(const QTypeRevision* param1) {
	 return new QTypeRevision(*param1);
}

void QTypeRevision_CopyAssign(QTypeRevision* self, QTypeRevision* other) {
    *self = *other;
}

void QTypeRevision_MoveAssign(QTypeRevision* self, QTypeRevision* other) {
    *self = std::move(*other);
}

QTypeRevision* QTypeRevision_Zero() {
	return new QTypeRevision(QTypeRevision::zero());
}

bool QTypeRevision_HasMajorVersion(const QTypeRevision* self) {
	return self->hasMajorVersion();
}

unsigned char QTypeRevision_MajorVersion(const QTypeRevision* self) {
	return self->majorVersion();
}

bool QTypeRevision_HasMinorVersion(const QTypeRevision* self) {
	return self->hasMinorVersion();
}

unsigned char QTypeRevision_MinorVersion(const QTypeRevision* self) {
	return self->minorVersion();
}

bool QTypeRevision_IsValid(const QTypeRevision* self) {
	return self->isValid();
}

void QTypeRevision_Delete(QTypeRevision* self) {
    delete self;
}

File: binding/include/libqtyperevision.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qtyperevision.html
pub const qtyperevision = struct {

    /// New constructs a new QTypeRevision object.
    pub fn New(other: ?*anyopaque) QtC.QTypeRevision {
        return qtc.QTypeRevision_new(@ptrCast(other));
    }


    /// New2 constructs a new QTypeRevision object.
    pub fn New2(other: ?*anyopaque) QtC.QTypeRevision {
        return qtc.QTypeRevision_new2(@ptrCast(other));
    }


    /// New3 constructs a new QTypeRevision object.
    pub fn New3() QtC.QTypeRevision {
        return qtc.QTypeRevision_new3();
    }


    /// New4 constructs a new QTypeRevision object.
    pub fn New4(param1: ?*anyopaque) QtC.QTypeRevision {
        return qtc.QTypeRevision_new4(@ptrCast(param1));
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QTypeRevision_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QTypeRevision_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn Zero() QtC.QTypeRevision {
        return qtc.QTypeRevision_Zero();
    }

    pub fn HasMajorVersion(self: ?*anyopaque, ) bool {
        return qtc.QTypeRevision_HasMajorVersion(@ptrCast(self));
    }

    pub fn MajorVersion(self: ?*anyopaque, ) u8 {
        return qtc.QTypeRevision_MajorVersion(@ptrCast(self));
    }

    pub fn HasMinorVersion(self: ?*anyopaque, ) bool {
        return qtc.QTypeRevision_HasMinorVersion(@ptrCast(self));
    }

    pub fn MinorVersion(self: ?*anyopaque, ) u8 {
        return qtc.QTypeRevision_MinorVersion(@ptrCast(self));
    }

    pub fn IsValid(self: ?*anyopaque, ) bool {
        return qtc.QTypeRevision_IsValid(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QTypeRevision_Delete(@ptrCast(self));
    }
};
